com.hybris.hyeclipse.editor/src/com/hybris/hyeclipse/editor/copyright/manager/CopyrightManager.java

```diff
@@ -0,0 +1,168 @@
+import com.hybris.hyeclipse.editor.preferences.CopyrightPreferenceConstants;
+import com.hybris.hyeclipse.hac.utils.ConsoleUtils;
+
+/**
+ * Class handling Copyright operations
+ */
+public class CopyrightManager {
+
+	private static final String ADD_COPYRIGHT = "Add Copyright";
+	private static final String OVERRIDE_COPYRIGHT = "Override Copyright";
+	private static final String NEW_LINE_SEPARATOR = "\n";
+
+	private ASTRewrite rewriter;
+	private CompilationUnitChange change;
+
+	/**
+	 * Gets copyright text from preference page
+	 *
+	 * @return copyright text
+	 */
+	public String getCopyrightText() {
+		final String[] contents = Activator.getDefault().getPreferenceStore()
+				.getString(CopyrightPreferenceConstants.COPYRIGHT_CONTENT).split(NEW_LINE_SEPARATOR);
+		final String firstLine = Activator.getDefault().getPreferenceStore()
+				.getString(CopyrightPreferenceConstants.COPYRIGHT_FIRST_LINE);
+		final String prefix = Activator.getDefault().getPreferenceStore()
+				.getString(CopyrightPreferenceConstants.COPYRIGHT_LINE_PREFIX);
+		final String lastLine = Activator.getDefault().getPreferenceStore()
+				.getString(CopyrightPreferenceConstants.COPYRIGHT_LAST_LINE);
+		final StringBuilder copyrightBuilder = new StringBuilder();
+		copyrightBuilder.append(firstLine + NEW_LINE_SEPARATOR);
+		for (final String line : contents) {
+			copyrightBuilder.append(prefix + line + NEW_LINE_SEPARATOR);
+		}
+		copyrightBuilder.append(lastLine + NEW_LINE_SEPARATOR);
+		return copyrightBuilder.toString();
+	}
+
+	/**
+	 * Adds copyright header to the compilation unit
+	 *
+	 * @param compilationUnit
+	 *            compilation unit affected
+	 * @return compilation unit change
+	 */
+	public CompilationUnitChange addCopyrightsHeader(final CompilationUnit compilationUnit) {
+		final ICompilationUnit unit = (ICompilationUnit) compilationUnit.getJavaElement();
+		change = new CompilationUnitChange(ADD_COPYRIGHT, unit);
+		rewriter = ASTRewrite.create(compilationUnit.getAST());
+		final ListRewrite listRewrite = rewriter.getListRewrite(compilationUnit.getPackage(),
+				PackageDeclaration.ANNOTATIONS_PROPERTY);
+		final Comment placeHolder = (Comment) rewriter.createStringPlaceholder(getCopyrightText(),
+				ASTNode.BLOCK_COMMENT);
+		listRewrite.insertFirst(placeHolder, null);
+		rewriteCompilationUnit(unit, getNewUnitSource(unit, null));
+		return change;
+	}
+
+	/**
+	 * Replaces copyright header to the compilation unit
+	 *
+	 * @param compilationUnit
+	 *            compilation unit affected
+	 * @return compilation unit change
+	 */
+	public CompilationUnitChange replaceCopyrightsHeader(final CompilationUnit compilationUnit) {
+		final ICompilationUnit unit = (ICompilationUnit) compilationUnit.getJavaElement();
+		change = new CompilationUnitChange(OVERRIDE_COPYRIGHT, unit);
+		rewriter = ASTRewrite.create(compilationUnit.getAST());
+		final List<Comment> comments = getCommentList(compilationUnit);
+		final Comment copyrightComment = comments.get(0);
```

Comment:
If there is no comment that means, the copyright header is not present so CopyrightFix will call another method of the CopyrightManager